src/value.rs

```diff
@@ -427,12 +440,23 @@ pub trait SetValue: StaticType {
     unsafe fn set_value(&mut Value, &Self);
 }
 
-impl FromValueOptional for String {
-    unsafe fn from_value_optional(value: &Value) -> Option<Self> {
+impl<'a> FromValueOptional<'a> for String {
+    unsafe fn from_value_optional(value: &'a Value) -> Option<Self> {
         from_glib_none(gobject_ffi::g_value_get_string(value.to_glib_none().0))
     }
 }
 
+impl<'a> FromValueOptional<'a> for &'a str {
+    unsafe fn from_value_optional(value: &'a Value) -> Option<Self> {
+        let cstr = gobject_ffi::g_value_get_string(value.to_glib_none().0);
+        if cstr.is_null() {
+            return None;
+        }
+
+        CStr::from_ptr(cstr).to_str().ok()
```

Comment:
Done :) Just a matter of taste I guess, I prefer to have the "error" code paths outside the normal code path.